sensirion_i2c.h

```diff
@@ -71,7 +71,7 @@ void sensirion_i2c_release(void);
  * @param count   number of bytes to read from I2C and store in the buffer
  * @returns 0 on success, error code otherwise
  */
-int8_t sensirion_i2c_read(uint8_t address, uint8_t *data, uint16_t count);
+int8_t sensirion_i2c_read(uint8_t address, uint8_t *data, uint8_t count);
```

Comment:
This changes the interface for all possible hardware backends, to match the one from one Arduino implementation. If you look at the problem from above, the problem is not the interface but the Arduino implementation. So that's where we'll need to start operating. Your diagnosis seems sane, but the fix is not. An appropriate fix would be to cast count into the appropriate type at the call location:

```cpp
Wire.requestFrom(address, (uint8_t)count);
```

This code alone has the downside that when count is > 256, the execution will silently truncate the output, so we'll never be able to read more than 256b without a fix on the caller side - not good since the current interface clearly expects this to be possible. So I see two possible fixes/workarounds:
1. we check at runtime with something like `if (count > 256) return ERR_TOO_MUCH_DATA;` causing all calls that attempt to send more than 256b of data to fail (this is the current behavior on your platform).
or better
2. we dynamically split the data in a for-loop in chunks of 256 bytes: `while (count > 0) { ... }`